File: Stage4/Python/Problem10810.py
def main():
    N, M = map(int, input().split())
    arr = [0] * N
    
    for _ in range(M):
        i, j, k = map(int, input().split())
        for index in range(i - 1, j):
            arr[index] = k
    
    print(*arr)
    
if __name__ == "__main__":
    main()

File: Stage4/C/Problem10810.c
#include <stdio.h>

int main() {
    int N, M;
    scanf("%d %d", &N, &M);
    int arr [N];
    int i, j, k;

    for (int x = 0; x < N; x++) {
        arr[x] = 0;
    }

    for (int x = 0; x < M; x++) {
        scanf("%d %d %d", &i, &j, &k);

        for (int index = i - 1; index <= j - 1; index++) {
            arr[index] = k;
        }
    }

    for (int x = 0; x < N; x++) {
        printf("%d ", arr[x]);
    }

    return 0;
}